### PR TITLE
docs: `integrations` reference updates 5

### PR DESCRIPTION
**Pull request by leo-gan**
Added missed references. Added missed provider pages.

### CODE REVIEW

File: docs/docs/integrations/providers/rank_bm25.mdx
Comment (baskaryan) [inline docs/docs/integrations/providers/rank_bm25.mdx:1]: not sure this qualifies as a provider

Comment (leo-gan) [reply docs/docs/integrations/providers/rank_bm25.mdx:1]: We have a good number of "Python package" providers. Examples: [Annoy](https://python.langchain.com/v0.2/docs/integrations/providers/annoy/), [ArcGIS](https://python.langchain.com/v0.2/docs/integrations/providers/arcgis/), [Beautiful Soup](https://python.langchain.com/v0.2/docs/integrations/providers/beautiful_soup/), [CTransformers](https://github.com/marella/ctransformers), [Doctran](https://github.com/psychic-api/doctran), [DSPy](https://python.langchain.com/v0.2/docs/integrations/providers/dspy/), ... it is only for [A-D].

If it is a package that specializes in the LLM tasks, it is a good candidate for a provider. If it is a generic package, like `requests` it is probably not a good candidate. This package, the `rank-bm25` is closely related to the LLM tasks.

**Comment by leo-gan** [comment]
@ccurme Can you review, please?

**Comment by leo-gan** [comment]
@ccurme Can you review, please?

File: docs/docs/integrations/providers/baidu.mdx [hunk 1]
```diff
@@ -39,6 +39,20 @@ See a [usage example](/docs/integrations/text_embedding/baidu_qianfan_endpoint).
 from langchain_community.embeddings import QianfanEmbeddingsEndpoint
 ```
 
+## Document loaders
+
+### Baidu BOS Directory Loader
+
+```python
+from langchain_community.document_loaders import BaiduBOSDirectoryLoader
```
Comment (ccurme) [inline docs/docs/integrations/providers/baidu.mdx:47]: looks like these are not exported from the `document_loaders` module.

**Review by ccurme — APPROVED** [review]

**Comment by leo-gan** [comment]
@ccurme Thanks for review!

**Pull request merged**